[ao][sparsity] Implemented the step() function

macandro96 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79883
* __->__ #79822
* #79817

The step() function calls the implemented get_schedule_param() and updates
the data_group dictionary of the data sparsifier.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataScheduler```

Differential Revision: [D37358609](https://our.internmc.facebook.com/intern/diff/D37358609)

Commit:
[WIP][ao][sparsity] Implemented the step() function for BaseDataScheduler class

TODO

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Implemented the step() function"


The step() function calls the implemented get_schedule_param() and updates
the data_group dictionary of the data sparsifier.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataScheduler```

[ghstack-poisoned]

jerryzh168 commented on test/ao/sparsity/test_data_scheduler.py:
```diff
@@ -38,12 +44,14 @@ def _get_data(self):
         return data_list, defaults, data_with_config
 
-    def _get_sparsifier(self):
+    def _get_sparsifier(self, return_data=False):
```
nit: if we pass around data in this api as suggested by the last PR we probably do not need this argument

Commit:
Update on "[ao][sparsity] Implemented the step() function"


The step() function calls the implemented get_schedule_param() and updates
the data_group dictionary of the data sparsifier.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataScheduler```

[ghstack-poisoned]

macandro96 commented on test/ao/sparsity/test_data_scheduler.py:
```diff
@@ -54,13 +62,63 @@ def _get_scheduler(self, sparsifier):
+                fname = '/'.join(fname.split('/')[-5:])
+                assert fname != 'torch/ao/sparsity/experimental/scheduler/data_scheduler/base_data_scheduler.py'
+
+    def test_step(self):
```
added the test for step_counter inside test_step()

macandro96 commented:
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37358609).

Commit:
Update on "[ao][sparsity] Implemented the step() function"


The step() function calls the implemented get_schedule_param() and updates
the data_group dictionary of the data sparsifier.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataScheduler```

Differential Revision: [D37358609](https://our.internmc.facebook.com/intern/diff/D37358609)

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Implemented the step() function"


The step() function calls the implemented get_schedule_param() and updates
the data_group dictionary of the data sparsifier.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataScheduler```

Differential Revision: [D37358609](https://our.internmc.facebook.com/intern/diff/D37358609)

[ghstack-poisoned]

macandro96 commented:
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37358609).
